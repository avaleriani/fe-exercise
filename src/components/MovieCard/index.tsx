import Link from "next/link";
import React, { FC } from "react";
import { formattedDate } from "../../../util/date";
import {
  ImageContainer,
  MovieCardWrapper,
  InfoContainer,
  LinkStyled,
  GenresWrapper,
  GenreStyled,
} from "./styled";

const MovieCard: FC<IMovieCard> = ({
  poster_path,
  title,
  genres,
  release_date,
  id,
}) => {
  return (
    <MovieCardWrapper data-testid="MovieCard__Component">
      <Link href={`/movie/${id}`}>
        <ImageContainer>
          <img src={poster_path} alt={title} />
        </ImageContainer>
      </Link>
      <InfoContainer>
        <LinkStyled href={`/movie/${id}`}>{title}</LinkStyled>
        <GenresWrapper>
          <GenreStyled>{genres?.toString()}</GenreStyled>
        </GenresWrapper>
        <div>{formattedDate(release_date)}</div>
      </InfoContainer>
    </MovieCardWrapper>
  );
};

export default MovieCard;
