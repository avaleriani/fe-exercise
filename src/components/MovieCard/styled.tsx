import Link from "next/link";
import styled from "styled-components";
import { size } from "../../../util/constants";

export const MovieCardWrapper = styled.div`
  margin: 10px;
  @media (min-width: ${size.mobileL}) {
    min-width: 250px;
    max-width: 250px;
  }
`;

export const ImageContainer = styled.a`
  width: 100%;
  border-radius: 10px;
  color: black;
  display: block;
  & > img {
    width: 100%;
    object-fit: contain;
    border-radius: 10px;
  }
`;

export const InfoContainer = styled.div`
  position: relative;
  padding: 15px 10px 10px 10px;
  color: white;
`;

export const LinkStyled = styled(Link)`
  white-space: nowrap;
  overflow: hidden;
  font-weight: bold;
  font-size: 18px;
  text-overflow: ellipsis;
  display: block;
`;

export const GenresWrapper = styled.div`
  margin: 16px 0;
`;

export const GenreStyled = styled.div`
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  font-size: 17px;
`;
