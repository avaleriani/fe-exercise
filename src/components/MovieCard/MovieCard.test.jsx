/**
 * @jest-environment jsdom
 */

import React from "react";

import { render, cleanup } from "@testing-library/react";

import MovieCard from "./index";

let getByTestId;
let component;

beforeEach(() => {
  component = render(<MovieCard />);
  getByTestId = component.getByTestId;
});

afterEach(() => {
  getByTestId = null;
  component = null;
  cleanup();
});

describe("MovieCard component", () => {
  it("should rendering on screen", () => {
    const actual = getByTestId("MovieCard__Component");
    expect(actual).toBeInTheDocument();
  });
});

it("renders character card unchanged", () => {
  const { container } = render(<MovieCard />);
  expect(container).toMatchSnapshot();
});
