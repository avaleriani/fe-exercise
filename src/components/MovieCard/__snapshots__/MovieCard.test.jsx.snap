// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders character card unchanged 1`] = `
<div>
  <div
    class="sc-c3766167-0 jDUJZn"
    data-testid="MovieCard__Component"
  >
    <a
      href="/movie/undefined"
    >
      <a
        class="sc-c3766167-1 kVSnzs"
      >
        <img />
      </a>
    </a>
    <div
      class="sc-c3766167-2 pdrtL"
    >
      <a
        class="sc-c3766167-3 dRGnkM"
        href="/movie/undefined"
      />
      <div
        class="sc-c3766167-4 gVywIT"
      >
        <div
          class="sc-c3766167-5 idgoDB"
        />
      </div>
      <div />
    </div>
  </div>
</div>
`;
